### PR TITLE
feat: My Properties and Property Editing

### PR DESCRIPTION
**Pull request by duolok**
This PR includes a component which lets host update information about any of his properties. It needs some updates on back-end to fully work.

This pull request contains  areas in the codebase that require refactoring. Refactoring efforts will be undertaken as time permits. 

_Preview_

image

image

image

### CODE REVIEW

File: src/app/properties/my-properties/my-properties.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  // approveProperty(property: Property): void {
+  //   this.propertyService.approveProperty(property)
+  //   .subscribe({
+  //       next:  (_) => {
+  //         this.getProperties();
+  //       }
+  //     })
+  // }
```
Comment (MomirMilutinovic) [inline src/app/properties/my-properties/my-properties.component.ts]: Remove commented code.
```suggestion
```

File: src/app/properties/property-edit/property-edit.component.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  menuItems: string[] = ['basic information', 'location', 'amenities', 'calendar', 'price table', 'other'];
+  propertyImages: File[] = [];
+  selectedMenuItem: string = this.menuItems[0];
+  availabilityEntries: PropertyAvailabilityEntry[] = [];
+  autoApproveEnabled: boolean = false;
+  cancellationDeadline: number = 0;
+  pricingMode: string = 'PER_PERSON';
+  id: number = 0;
+  propertyNameControl: FormControl = new FormControl('', Validators.required);
+  dateRange: DateRange<Date> = new DateRange<Date>(this.beginDate, this.endDate);
```
Comment (MomirMilutinovic) [inline src/app/properties/property-edit/property-edit.component.ts]: There are several unused variables here.
```suggestion
    menuItems: string[] = ['basic information', 'location', 'amenities', 'calendar', 'price table', 'other'];
  propertyImages: File[] = [];
  selectedMenuItem: string = this.menuItems[0];
  id: number = 0;
  propertyNameControl: FormControl = new FormControl('', Validators.required);
```

File: src/app/properties/property-updated-dialog/property-updated-dialog.component.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+h1 {
+  font-family: bayon;
+  text-align: center;
+  padding-top: 4%;
+  padding-bottom: 4%;
+  font-size: 3rem;
+}
+span {
+  color : #00B3B3 !importnat;
```
Comment (MomirMilutinovic) [inline src/app/properties/property-updated-dialog/property-updated-dialog.component.css]:
```suggestion
  color : #00B3B3 !important;
```

**Review by MomirMilutinovic — CHANGES_REQUESTED** [review]
Looks good. I only have some refactoring suggestions.

File: src/app/properties/property.service.ts [hunk 4]
```diff
@@ -44,11 +44,15 @@ export class PropertyService {
 
   getUnapprovedProperties(searchDto?: PropertiesSearchDto): Observable<Property[]> {
     let url = `${environment.apiHost}/${ApiPaths.UnapprovedProperties}`;
-
-
     return this.http.get<Property[]>(url);
   }
 
+  getMyProperties(hostId: any): Observable<Property[]> {
+    let url = `${environment.apiHost}/${ApiPaths.Properties}/${ApiPaths.MyProperties}`;
```
Comment (MomirMilutinovic) [inline src/app/properties/property.service.ts:51]: It seems that this endpoint does not exist on the backend.

Comment (duolok) [reply src/app/properties/property.service.ts:51]: > It needs some updates on back-end to fully work.

This explains it.

Comment (MomirMilutinovic) [reply src/app/properties/property.service.ts:51]: Ok, sorry about that.

**Comment by duolok** [comment]
This should work now.

**Review by MomirMilutinovic — COMMENTED** [review]
Looks good.

**Pull request merged**